### PR TITLE
Fix exclude filter

### PR DESCRIPTION
**Pull request by TheOneRing**

### CODE REVIEW

File: src/csync/csync_exclude.cpp [hunk 1]
```diff
@@ -361,6 +361,14 @@ bool ExcludedFiles::isExcludedRemote(const QString &filePath, const QString &bas
         relativePath.chop(1);
     }
 
+    if (excludeHidden) {
+        // Check ther parent directories for . hidden files
```
Comment (fmoc) [inline src/csync/csync_exclude.cpp:365]:
```suggestion
        // Check their parent directories for . hidden files
```

File: src/csync/csync_exclude.cpp [hunk 2]
```diff
@@ -361,6 +361,14 @@ bool ExcludedFiles::isExcludedRemote(const QString &filePath, const QString &bas
...
+        if (relativePath.startsWith(QLatin1Char('.')) || relativePath.contains(QLatin1String("/."))) {
```
Comment (fmoc) [inline src/csync/csync_exclude.cpp:366]: Without seeing an example value for `relativePath`, I can't tell whether these checks make sense. I would expect you to calculate the base directory name first before using `startsWith`.
Perhaps the variable name should be updated?

Comment (TheOneRing) [reply src/csync/csync_exclude.cpp:366]: The relativePath looks like `foo/bar` or `.foo/.bar`

**Review by fmoc — APPROVED** [review]

**Comment by TheOneRing** [comment]
As the unit test for this runs out of mem.....

**Comment by fmoc** [comment]
Weird, I don't see any obvious chance for a memleak in the code... the test just seem to fail on Windows.

**Review by dschmidt — APPROVED** [review]

**Comment by TheOneRing** [comment]
I can imagine its a recursion issue..

**Comment by TheOneRing** [comment]
CI:
```
 10/38 Test #11: ExcludedFilesTest ................***Failed    0.52 sec
isExcludedC:/Users/runneradmin/AppData/Local/Temp/qt_temp-dkuPmF/a/b C:/Users/runneradmin/AppData/Local/Temp/qt_temp-dkuPmF/a
isExcludedC:/Users/runneradmin/AppData/Local/Temp/qt_temp-BoUEaa/a/b~ C:/Users/runneradmin/AppData/Local/Temp/qt_temp-BoUEaa/a
isExcludedC:/Users/runneradmin/AppData/Local/Temp/qt_temp-EUTOpq/a/.b C:/Users/runneradmin/AppData/Local/Temp/qt_temp-EUTOpq/a
isExcludedC:/Users/runneradmin/AppData/Local/Temp/qt_temp-qRffDK/a/.b C:/Users/runneradmin/AppData/Local/Temp/qt_temp-qRffDK/a
isExcludedC:/Users/runneradmin/AppData/Local/Temp/qt_temp-jjjSPq/a/b C:/Users/runneradmin/AppData/Local/Temp/qt_temp-jjjSPq/a
isExcludedC:/Users/runneradmin/AppData/Local/Temp/qt_temp-EfjWJq/a/b~ C:/Users/runneradmin/AppData/Local/Temp/qt_temp-EfjWJq/a
isExcludedC:/Users/runneradmin/AppData/Local/Temp/qt_temp-jWioGk/a/.b C:/Users/runneradmin/AppData/Local/Temp/qt_temp-jWioGk/a
isExcludedC:/Users/runneradmin/AppData/Local/Temp/qt_temp-RYhHxO/a/.Trashes C:/Users/runneradmin/AppData/Local/Temp/qt_temp-RYhHxO/a
isExcludedC:/Users/runneradmin/AppData/Local/Temp/qt_temp-WXkDuN/a/foo_conflict-bar C:/Users/runneradmin/AppData/Local/Temp/qt_temp-WXkDuN/a
isExcludedC:/Users/runneradmin/AppData/Local/Temp/qt_temp-OqKywK/a/foo (conflicted copy bar) C:/Users/runneradmin/AppData/Local/Temp/qt_temp-OqKywK/a
isExcludedC:/Users/runneradmin/AppData/Local/Temp/qt_temp-NxUWVK/a/.b C:/Users/runneradmin/AppData/Local/Temp/qt_temp-NxUWVK/a
isExcluded/a/.b /a
isExcluded/a /a
```

Vs local:
```
Config: Using QtTest library 5.12.10, Qt 5.12.10 (x86_64-little_endian-llp64 shared (dynamic) debug build; by MSVC 2017)
PASS   : TestExcludedFiles::initTestCase()
isExcludedC:/Users/hanna/AppData/Local/Temp/qt_temp-hSEbTW/a/b C:/Users/hanna/AppData/Local/Temp/qt_temp-hSEbTW/a
isExcludedC:/Users/hanna/AppData/Local/Temp/qt_temp-ujDaEE/a/b~ C:/Users/hanna/AppData/Local/Temp/qt_temp-ujDaEE/a
isExcludedC:/Users/hanna/AppData/Local/Temp/qt_temp-RyYWoc/a/.b C:/Users/hanna/AppData/Local/Temp/qt_temp-RyYWoc/a
isExcludedC:/Users/hanna/AppData/Local/Temp/qt_temp-bxnxcC/a/.b C:/Users/hanna/AppData/Local/Temp/qt_temp-bxnxcC/a
isExcludedC:/Users/hanna/AppData/Local/Temp/qt_temp-dImXpb/a/b C:/Users/hanna/AppData/Local/Temp/qt_temp-dImXpb/a
isExcludedC:/Users/hanna/AppData/Local/Temp/qt_temp-XCnpgK/a/b~ C:/Users/hanna/AppData/Local/Temp/qt_temp-XCnpgK/a
isExcludedC:/Users/hanna/AppData/Local/Temp/qt_temp-vzdrxY/a/.b C:/Users/hanna/AppData/Local/Temp/qt_temp-vzdrxY/a
isExcludedC:/Users/hanna/AppData/Local/Temp/qt_temp-YghpVH/a/.Trashes C:/Users/hanna/AppData/Local/Temp/qt_temp-YghpVH/a
isExcludedC:/Users/hanna/AppData/Local/Temp/qt_temp-nvpZBz/a/foo_conflict-bar C:/Users/hanna/AppData/Local/Temp/qt_temp-nvpZBz/a
isExcludedC:/Users/hanna/AppData/Local/Temp/qt_temp-jOKydF/a/foo (conflicted copy bar) C:/Users/hanna/AppData/Local/Temp/qt_temp-jOKydF/a
isExcludedC:/Users/hanna/AppData/Local/Temp/qt_temp-WDOunI/a/.b C:/Users/hanna/AppData/Local/Temp/qt_temp-WDOunI/a
isExcluded/a/.b /a
isExcluded/a /a
isExcluded/ /a
```
The question why we fail remains.

**Comment by TheOneRing** [comment]
Output fixed in https://invent.kde.org/packaging/craft/-/commit/bc713fea69acd9f9f2ccf9d8239f10c94a94eac8
With this change we take exclude of the parent folder for non existing files.
On the github ci the folder /a/ happened to exists on Windows which caused the failing unit test.

**Pull request merged**